pkg/azuredisk/nodeserver_v2.go

```diff
@@ -85,14 +85,16 @@ func (d *DriverV2) NodeStageVolume(ctx context.Context, req *csi.NodeStageVolume
 			return nil, status.Errorf(codes.Internal, "Failed to get perf attributes for %s. Error: %v", source, err)
 		}
 
-		if d.getDeviceHelper().DiskSupportsPerfOptimization(profile, accountType) {
-			err = d.getDeviceHelper().OptimizeDiskPerformance(d.getNodeInfo(), d.getDiskSkuInfoMap(), source, profile, accountType,
-				diskSizeGibStr, diskIopsStr, diskBwMbpsStr)
-			if err != nil {
-				return nil, status.Errorf(codes.Internal, "Failed to optimize device performance for target(%s) error(%s)", source, err)
+		if isPerfTuningEnabled(profile) {
+			// save one logging since perfTuning is disabled by default
+			if d.getDeviceHelper().DiskSupportsPerfOptimization(profile, accountType) {
+				if err := d.getDeviceHelper().OptimizeDiskPerformance(d.getNodeInfo(), d.getDiskSkuInfoMap(), source, profile, accountType,
+					diskSizeGibStr, diskIopsStr, diskBwMbpsStr); err != nil {
+					return nil, status.Errorf(codes.Internal, "Failed to optimize device performance for target(%s) error(%s)", source, err)
+				}
+			} else {
+				klog.V(2).Infof("NodeStageVolume: perf optimization is disabled for %s. perfProfile %s accountType %s", source, profile, accountType)
 			}
-		} else {
-			klog.V(2).Infof("NodeStageVolume: perf optimization is disabled for %s. perfProfile %s accountType %s", source, profile, accountType)
```

Comment:
This is a single log line per attach which tells me if all the properties made it to volume attributes. Will allow us to easily confirm if the perf optimization was not done on disk. I think we should leave it.
Absence of logs are not always a good confirmation while investigating issues.

Reply:
`isPerfTuningEnabled(profile)` only checks whether `profile != none` (`none` is by default), and this `profile` setting is already logged in `NodeStageVolume` request body, no need to log here again.

Reply:
if one feature is not enabled in storage class by user, no need to mention again in logs

Reply:
I mean if it's `"perfProfile":"None"`, no need to print log again since it's not enabled; while if it's `"perfProfile":"Basic"` and we should definitely print logs no matter if optimization happens or not.

```
I0528 03:29:16.345597       1 utils.go:95] GRPC call: /csi.v1.Node/NodeStageVolume
I0528 03:29:16.345617       1 utils.go:96] GRPC request: {"publish_context":{"LUN":"0"},"staging_target_path":"/var/lib/kubelet/plugins/kubernetes.io/csi/pv/azuredisk-2079-disk.csi.azure.com-preprovsioned-pv-cd8vs/globalmount","volume_capability":{"AccessType":{"Mount":{}},"access_mode":{"mode":1}},"volume_context":{"cachingMode":"None","maxShares":"5","perfProfile":"None","requestedsizegib":"1024","skuName":"Premium_LRS"},"volume_id":"/subscriptions/0e46bd28-a80f-4d3a-8200-d9eb8d80cb2e/resourceGroups/kubetest-dwgypo97/providers/Microsoft.Compute/disks/shared-disk-multiple-pods"}
I0528 03:29:18.071367       1 azure_common_linux.go:175] azureDisk - found /dev/disk/azure/scsi1/lun0 by sdc under /dev/disk/azure/scsi1/
I0528 03:29:18.071402       1 nodeserver.go:109] NodeStageVolume: perf optimization is disabled for /dev/disk/azure/scsi1/lun0. perfProfile None accountType Premium_LRS
```

Reply:
anyway, I reverted, let's move forward.